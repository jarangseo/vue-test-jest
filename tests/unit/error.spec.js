import getErrorMessage from '@/shared/utils/error'

describe("getErrorMessage", () => {
  it("returns camel message when code is 1", () => {
    expect(getErrorMessage(1)).toBe("The camel walks on a leg");
  });

  it("returns rabbit message when code is 2", () => {
    expect(getErrorMessage(2)).toBe("Rabbits don't eat carrots");
  });

  it("returns cat message when code is 3", () => {
    expect(getErrorMessage(3)).toBe("Cats don't eat mouses");
  });

  it("throws an error otherwise", () => {
    expect(() => getErrorMessage(4)).toThrow("No error messages for that code");
  });
});

describe("getErrorMessage with Snapshot", () => {
  it("returns an error for a valid code", () => {
    expect(getErrorMessage(1)).toMatchSnapshot();
    expect(getErrorMessage(2)).toMatchSnapshot();
    expect(getErrorMessage(3)).toMatchSnapshot();
  });

  it("throws an error otherwise", () => {
    expect(() => getErrorMessage(4)).toThrowErrorMatchingSnapshot();
  });
});