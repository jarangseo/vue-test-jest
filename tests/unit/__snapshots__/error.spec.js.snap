// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getErrorMessage with Snapshot returns an error for a valid code 1`] = `"The camel walks on a leg"`;

exports[`getErrorMessage with Snapshot returns an error for a valid code 2`] = `"Rabbits don't eat carrots"`;

exports[`getErrorMessage with Snapshot returns an error for a valid code 3`] = `"Cats don't eat mouses"`;

exports[`getErrorMessage with Snapshot throws an error otherwise 1`] = `"No error messages for that code"`;
